clients/src/test/java/org/apache/kafka/clients/producer/internals/BufferPoolTest.java

```diff
@@ -171,14 +181,14 @@ public void testBlockTimeout() throws Exception {
         try {
             pool.allocate(10, maxBlockTimeMs);
             fail("The buffer allocated more memory than its maximum value 10");
-        } catch (TimeoutException e) {
+        } catch (BufferExhaustedException e) {
```

Comment:
Similar to the above, perhaps reduce maxBlockTimeMs to 10ms?

Reply:
thanks @junrao - I answered in your related comment above.